pkg/allocator/allocator_test.go

```diff
@@ -226,7 +226,7 @@ func (t TestAllocatorKey) Value(any) any {
 }
 
 func randomTestName() string {
-	return rand.RandomStringWithPrefix(testPrefix, 12)
+	return fmt.Sprintf("%s%s", testPrefix, rand.String(12))
```

Comment:
Nit: I'm wondering if it might be worth to use a `+` here instead of the `fmt.Sprintf` (see https://github.com/cilium/cilium/issues/19571). I don't know if this qualifies as a hot path, if not, feel free to ignore this comment.

Reply:
@pippolo84 this is a unit test so I'm not that concerned about it.